### DISCUSSION TITLE
installing the library

### DISCUSSION THREAD

**Discussion by CazyKiid** [D7EAD/liboai#48] [Q&A] [answered]
Hy guys Im new to C++ and Visual Studio. I dont really know how to install or make use of library and I'm finding it hard to install the  liboai library because there are no clear in instructions. I already installed curl and json... When i copy the includes to my project and cpp files they dont work correctly. Please help me install the library

**Comment by asnagni** [comment, marked as answer]
Hi,
  I work on a Mac computer and I don't know much about Windows but I guess the process should be same:
  1) Make sure that you have curl library installed on your computer (the latest version will be good)
  2) Make sure to include the library (I guess .lib on Windows) and the include the header files into your project
  2) Include the single nlohmann json file into your project like you will do with any other file of your project.
  3) After that you need to include the liboai into your project. This is very simple. I included the files like any other files (.hpp or .cpp) into my project. My advise is to keep the original struct and integrate it into your project. No need to include the examples.
  
  To help you, I attached a screenshot of my setup with Xcode. You will see the liboai included into my project. You just need to map that to whatever tool you are using on Windows.
  
![image](https://github.com/D7EAD/liboai/assets/37470668/7e13a53d-f8b0-422e-9299-84e5fa519fcc)

  Thanks you,
  Stay safe

**Comment by D7EAD** [comment]
Note that the above answer mentions a library file for `liboai`. The library no longer uses static or dynamic linkage for quite some time, and can quite literally just be added to your codebase via the .cpp and .h files., assuming C++17 compliance.

The files and proper linking are still required for nlohmann (pure C++, so no linkage) and cURL.
